driver/bindings/junit/src/main/scala/org/virtuslab/ideprobe/DSL.scala

```diff
@@ -9,6 +9,7 @@ final class RunningIntelliJFixture(
     val workspace: Path,
     val probe: ProbeDriver,
     val config: Config,
+    val test: TestConfig,
```

Comment:
should I implement the implicit `ConfigReader` for the `TestConfig` or alternatively simply utilising something like the `get[Seq[String]]("test.modules")` on the final `Config` will suffice?

Reply:
Implicit reader should be OK. It will simplify the call-site

Reply:
Each test usually has different kind properties, and config is designed and used in more ad-hoc way.
I would find it simpler eg. for such config:
```
project {
  modules = [a, b, c]
  name = x
}
```

to have just
```
val projectName = intelliJ.config[String]("project.name")
val projectName = intelliJ.config[Seq[String]]("project.modules")
```

instead of 

```
class Test extends ConfigFormat {
  case class TestProject(name: String, modules: Seq[String]
  import pureconfig.generic.auto._
  val expectedProject = intelliJ.config[TestProject]("project")
}
```

this is for the case when we have primitives in lists. I would suggest otherwise when we deal with lists of structures:
```
project {
  modules = [
    { name = a, contentRoots = [] },
    { name = b, contentRoots = [x/y] }
  ]
}
```

In such case it is worth to pull out the advanced apis to generate config and use it for easier comparison, etc. as it would be simpler than alternatives.

To sum up, I'd always go with what is more simple and straightforward.